src/full_coverage_path_planner.cpp

```diff
@@ -191,17 +191,18 @@ namespace full_coverage_path_planner
     RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Plan ready containing %lu goals!", plan.size());
   }
 
-  bool FullCoveragePathPlanner::parseGrid(nav_msgs::msg::OccupancyGrid const &cpp_grid_,
+  bool FullCoveragePathPlanner::parseGrid(nav2_costmap_2d::Costmap2D const * cpp_costmap,
                                           std::vector<std::vector<bool>> &grid,
                                           float robotRadius,
                                           float toolRadius,
                                           geometry_msgs::msg::PoseStamped const &realStart,
                                           Point_t &scaledStart)
   {
     uint ix, iy, nodeRow, nodeColl;
-    uint32_t nodeSize = dmax(floor(toolRadius / cpp_grid_.info.resolution), 1);       // Size of node in pixels/units
-    uint32_t robotNodeSize = dmax(floor(robotRadius / cpp_grid_.info.resolution), 1); // RobotRadius in pixels/units
-    uint32_t nRows = cpp_grid_.info.height, nCols = cpp_grid_.info.width;
+    uint32_t nodeSize = dmax(floor(toolRadius / cpp_costmap->getResolution()), 1);       // Size of node in pixels/units
+    uint32_t robotNodeSize = dmax(floor(robotRadius / cpp_costmap->getResolution()), 1); // RobotRadius in pixels/units
+    uint32_t nRows = cpp_costmap->getSizeInCellsY(), nCols = cpp_costmap->getSizeInCellsX();
```

Comment:
May I now nitpick about all declerations on it's own line? ;-)

Reply:
you may ;) I also changed to type size_t in this function